faker/providers/internet/__init__.py

```diff
@@ -178,6 +210,45 @@ def ipv4(self, network=False):
             address = str(ip_network(address, strict=False))
         return address
 
+    def ipv4_network_class(self):
+        return self.random_element('abc')
+
+    def ipv4_private(self, address_class=None, network=False):
```

Comment:
eventually I want to remove `ipv4_private` and `ipv4_public` from the api in favor of a `private=None` argument for `ipv4`

Reply:
n/m this comment, I've read a little bit and know I understand that private networks are in multiple classes

Reply:
I think the best way to go about this is to have:
```python
def ipv4(self, network=False, address_class=None, private=None):
    if address_class is not None or private is not None:
        if private is True:
            return self.ipv4_private(address_class=address_class, network=network)
        else:
            return self.ipv4_public(address_class=address_class, network=network)
    # old logic here...
```